utils/litemap/src/map.rs

```diff
@@ -192,6 +197,38 @@ impl<K: Ord, V> LiteMap<K, V> {
             Err(_) => None,
         }
     }
+
+    /// Extend the map's underlying vector with the items in the iterator as long as
+    /// the appended keys remain in sorted order with the vector's existing keys.
+    ///
+    /// If successful, return the map containing the appended items.
+    /// Otherwise return the map and the rest of the unsorted items.
```

Comment:
I was expecting this function to return `Result<Self, Error>` rather than `Result<Self, (Self, E)>`.  I think this works, but my expectation is that most people who call this function should expect their data to be well-formed, and if it's not, then something is wrong and they should exit.  If they expect their data to be unsorted, then they should use the unsorted method.

It's also a little weird that you have one single `.insert()` in this function.